src/queue.cpp

```diff
@@ -787,6 +787,20 @@ cvk_command_kernel::build_batchable_inner(cvk_command_buffer& command_buffer) {
         return err;
     }
 
+    // The queue has a cvk_printf_handler which will hold on to the cvk_mem of
+    // the printf buffer and process it after the queue has finished
+    if (m_kernel->program()->uses_printf()) {
+        // Zero the printf buffer
+        auto buffer = m_argument_values->get_printf_buffer();
+        std::vector<char> zeroes = {};
+        size_t size = buffer->size();
+        zeroes.resize(size);
+        memset(zeroes.data(), 0, size);
+        if (buffer->copy_from(zeroes.data(), 0, size))
```

Comment:
Any reason you're not using memset directly on a host-mapped pointer to the buffer's memory? The way it's done looks very inefficient. Taking a step back: do you really need to zero the entire buffer? With the default size, we're talking about 1MB's worth of zeroing for each kernel instance! Am I missing something?

Reply:
No good reason for not using memset on the mapped memory directly, I've fixed that now. An older version did require the whole buffer to be zeroed, but that's not the case any more. I've updated this to only zero the first 4 bytes (i.e. the allocation counter).

Reply:
Looks good, resolving.